python/torch_mlir_e2e_test/torchscript/framework.py

```diff
@@ -238,40 +269,89 @@ def generate_golden_trace(test: Test) -> Trace:
     return trace
 
 
-def run_tests(tests: List[Test], config: TestConfig) -> List[TestResult]:
-    """Invoke the given `Test`'s with the provided `TestConfig`."""
-    results = []
-    for test in tests:
-        # TODO: Precompile everything in parallel.
-        try:
-            golden_trace = generate_golden_trace(test)
-            compiled = config.compile(test.program_factory())
-        except Exception as e:
-            results.append(
-                TestResult(unique_name=test.unique_name,
-                           compilation_error="".join(traceback.format_exception(
-                               type(e), e, e.__traceback__)),
-                           runtime_error=None,
-                           trace=None,
-                           golden_trace=None))
-            continue
-        # TODO: Run in parallel.
-        try:
-            trace = config.run(compiled, golden_trace)
-        except Exception as e:
-            results.append(
-                TestResult(unique_name=test.unique_name,
-                           compilation_error=None,
-                           runtime_error="".join(traceback.format_exception(
-                               type(e), e, e.__traceback__)),
-                           trace=None,
-                           golden_trace=None))
-            continue
+def compile_and_run_test(test_name: str, results: mp.Queue, tests: Dict[str, Test],
```

Comment:
nit: can you change this function to have a signature `compile_and_run_test(test: Test, config: TestConfig) -> TestResult` (that knows nothing about parallelism/etc.? And then inline the other parts of this function into `def worker` below.